cpp/box/Box.h

```diff
@@ -381,6 +382,53 @@ class Box
         });
     }
 
+    //! Calculate distance between two points using boundary conditions
+    /*! \param p_i Particle position
+        \param p_j Particle position to calculate distances from
+    */
+    float computeDistance(const vec3<float>& p_i, const vec3<float>& p_j) const
+    {
+            vec3<float> v_ij = wrap(p_j - p_i);
+            return std::sqrt(dot(v_ij, v_ij));
+    }
+
+    //! Calculate distance between a set of points and query points using boundary conditions
+    /*! \param points Particle positions
+        \param query_points Particle position to calculate distances from
+        \param dist Distances between points and query_points
+        \param Nvecs The number of points to calculate distances between
+    */
+    void computeDistances(vec3<float>* points, vec3<float>* query_points, float *dist, unsigned int Nvecs) const
```

Comment:
The input arrays can be `const`. Also I prefer `distances` over `dist` to make it clear that it's plural (array-like).
```suggestion
    void computeDistances(const vec3<float>* points, const vec3<float>* query_points, float* distances, const unsigned int Nvecs) const
```